earthfile2llb/earthfile2llb.go

```diff
@@ -115,6 +115,9 @@ func Earthfile2LLB(ctx context.Context, target domain.Target, opt ConvertOpt) (m
 	if err != nil {
 		return nil, err
 	}
+	if err := validateAst(bc.Earthfile); err != nil {
+		return nil, err
+	}
```

Comment:
I would put this check in the build context resolver, right after the parse call. That way  `bc` always has a validated `bc.Earthfile`.

Reply:
How about putting it in the `Parse` method itself? Then its always valid after parsing, and we don't have to make validation public?

Reply:
That might be better